schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -1,549 +1,835 @@
     },
-    "definitions": {
-        "id": {
-            "properties": {
-                "id": {
-                    "type": "string"
-                }
-            },
-            "type": "object",
-            "required": [
-                "id"
-            ]
+    "extensions": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.Compute/virtualMachines/extensions"
+          ]
         },
-        "networkInterfaces": {
-            "properties": {
-                "id": {
-                    "type": "string"
-                },
-                "properties": {
-                    "properties": {
-                        "primary": {
-                            "type": "boolean"
-                        }
-                    },
-                    "type": "object",
-                    "required": [
-                        "primary"
-                    ]
-                }
+        "apiVersion": {
+          "enum": [
+            "2015-05-01-preview",
+            "2015-06-15"
+          ]
+        },
+        "properties": {
+          "properties": {
+            "publisher": {
+              "type": "string"
             },
-            "type": "object",
-            "required": [
-                "id"
-            ]
+            "type": {
+              "type": "string"
+            },
+            "typeHandlerVersion": {
+              "type": "string"
+            },
+            "settings": {
+              "type": "object"
+            }
+          },
+          "type": "object",
+          "required": [
+            "publisher",
+            "type",
+            "typeHandlerVersion",
+            "settings"
+          ]
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "extensionsChild": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "extensions"
+          ]
         },
-        "hardwareProfile": {
-            "properties": {
-                "vmSize": {
-                    "type": "string"
-                }
+        "apiVersion": {
+          "enum": [
+            "2015-05-01-preview",
+            "2015-06-15"
+          ]
+        },
+        "properties": {
+          "properties": {
+            "publisher": {
+              "type": "string"
             },
-            "type": "object",
-            "required": [
-                "vmSize"
-            ]
+            "type": {
+              "type": "string"
+            },
+            "typeHandlerVersion": {
+              "type": "string"
+            },
+            "settings": {
+              "type": "object"
+            }
+          },
+          "type": "object",
+          "required": [
+            "publisher",
+            "type",
+            "typeHandlerVersion",
+            "settings"
+          ]
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "sku": {
```

Comment:
all the new objects except "virtualMachineScaleSets" are not resource definition, please move them under "definitions"